homeassistant/components/sleepiq/sensor.py

```diff
@@ -17,27 +24,27 @@ async def async_setup_entry(
     """Set up the SleepIQ bed sensors."""
     coordinator: SleepIQDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
     async_add_entities(
-        SleepNumberSensor(coordinator, bed_id, side)
-        for side in SIDES
-        for bed_id in coordinator.data
-        if getattr(coordinator.data[bed_id][BED], side) is not None
+        SleepNumberSensorEntity(coordinator, bed, sleeper)
+        for bed in coordinator.client.beds.values()
+        for sleeper in bed.sleepers
     )
 
 
-class SleepNumberSensor(SleepIQSensor, SensorEntity):
-    """Implementation of a SleepIQ sensor."""
+class SleepNumberSensorEntity(SleepIQSensor, SensorEntity):
+    """Representation of an SleepIQ Entity with CoordinatorEntity."""
+
+    _attr_icon = ICON
```

Comment:
```suggestion
    _attr_icon = "mdi:bed"
```

If you are only using it one place, its ok to not make it a constant. Either way is fine though